core/src/main/java/org/mule/transport/polling/PollingMessageSource.java

```diff
@@ -0,0 +1,421 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.transport.polling;
+
+import org.mule.DefaultMuleEvent;
+import org.mule.DefaultMuleMessage;
+import org.mule.MessageExchangePattern;
+import org.mule.OptimizedRequestContext;
+import org.mule.VoidMuleEvent;
+import org.mule.api.MessagingException;
+import org.mule.api.MuleContext;
+import org.mule.api.MuleEvent;
+import org.mule.api.MuleException;
+import org.mule.api.MuleMessage;
+import org.mule.api.construct.FlowConstruct;
+import org.mule.api.construct.FlowConstructAware;
+import org.mule.api.context.MuleContextAware;
+import org.mule.api.context.WorkManager;
+import org.mule.api.execution.ExecutionCallback;
+import org.mule.api.execution.ExecutionTemplate;
+import org.mule.api.lifecycle.CreateException;
+import org.mule.api.lifecycle.Disposable;
+import org.mule.api.lifecycle.Initialisable;
+import org.mule.api.lifecycle.InitialisationException;
+import org.mule.api.lifecycle.Startable;
+import org.mule.api.lifecycle.Stoppable;
+import org.mule.api.processor.MessageProcessor;
+import org.mule.api.schedule.Scheduler;
+import org.mule.api.schedule.SchedulerFactory;
+import org.mule.api.source.MessageSource;
+import org.mule.config.i18n.CoreMessages;
+import org.mule.context.notification.ConnectorMessageNotification;
+import org.mule.execution.TransactionalErrorHandlingExecutionTemplate;
+import org.mule.transport.NullPayload;
+import org.mule.transport.TrackingWorkManager;
+import org.mule.util.StringUtils;
+import org.mule.util.concurrent.NamedThreadFactory;
+import org.mule.util.concurrent.ThreadNameHelper;
+
+import java.util.Map;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+//MULE-9139 - TODO verify if only one connector was created for all the polling elements in the config. If that's the case then we need to create a centralized service for the scheduling service
```

Comment:
This comment does not apply anymore. After I wrote this I realized that the whole connector is not used at all. Only the scheduling service is used.